wagtail/models/__init__.py

```diff
@@ -673,6 +838,48 @@ def check(cls, **kwargs):
 
         return errors
 
+    def log_edited(self, revision):
+        logger.info(
+            'Page edited: "%s" id=%d revision_id=%d',
+            self.title,
+            self.id,
+            revision.id,
+        )
+
+    def log_submitted_for_moderation(self, revision):
+        logger.info(
+            'Page submitted for moderation: "%s" id=%d revision_id=%d',
+            self.title,
+            self.id,
+            revision.id,
+        )
+
+    def save_revision(
+        self,
+        user=None,
+        submitted_for_moderation=False,
+        approved_go_live_at=None,
+        changed=True,
+        log_action=False,
+        previous_revision=None,
+        clean=True,
+    ):
+        # Raise an error if this page is an alias.
+        if self.alias_of_id:
+            raise RuntimeError(
+                "save_revision() was called on an alias page. "
+                "Revisions are not required for alias pages as they are an exact copy of another page."
+            )
+        return super().save_revision(
```

Comment:
Thoughts on putting this here instead of in `Revision`? This one's a bit easier as it's right at the beginning of the method.

If we decide to add the hooks, might be better to put this in the before hook.